### PR TITLE
Update Migration Guide: Unsafe Marker

### PR DESCRIPTION
**Pull request by khajavi**

### CODE REVIEW

File: docs/guides/migrate/migration-guide.md [hunk 1]
```diff
@@ -1262,7 +1354,7 @@ val app: ZIO[Any, Nothing, Unit] =
 
 > Note:
 > 
-> In ZIO 2.x, the `ZIO#provide` method—together with its variant `ZIO#provideSome`—is a default and easier way of injecting dependencies to the environmental effect. We do not require creating the dependency graph manually, it will be automatically generated. In contrast, the `ZIO#provideLayer`—and its variant `ZIO#provideSomeLayer`—is useful for low-level and custom cases like.
+> In ZIO 2.x, the `ZIO#provide` method—and all its variants `ZIO#provideSome`, `ZIO#provideCustom`—is a default and easier way of injecting dependencies to the environmental effect. We do not require creating the dependency graph manually, it will be automatically generated. In contrast, the `ZIO#provideLayer`—and all its variants `ZIO#provideSomeLayer`, `ZIO#provideCustomLayer`—is useful for low-level and custom cases like.
```
Comment (adamgfraser) [inline docs/guides/migrate/migration-guide.md]: I think this is an unnecessary change. There was another PR that we merged to delete references to `provideCustom` since it was deleted.

File: docs/guides/migrate/migration-guide.md [hunk 2]
```diff
@@ -729,6 +729,98 @@ We deprecated the `Fiber.ID` and moved it to the `zio` package and called it the
 
 ## Runtime, Platform and Executor
 
+## Unsafe Marker
+
+To run a ZIO workflow, we usually use `ZIOAppDefault` or `ZIOApp` traits. These traits provide the `run` method which will run the workflow using their default `Runtime` system. But when we want to work with a low-level API or want to integrate with a legacy code, we need to unsafely run the workflow.
+
+In ZIO 1.x, we used the `zio.Runtime.unsafeRun` method to run a ZIO workflow:
+
+```scala
```
Comment (adamgfraser) [inline docs/guides/migrate/migration-guide.md:738]: Would be good to use `mdoc` on these examples where possible so we test that they compile.

File: docs/guides/migrate/migration-guide.md [hunk 3]
```diff
@@ -729,6 +729,98 @@ We deprecated the `Fiber.ID` and moved it to the `zio` package and called it the
...
+trait Runtime[+R] {
+  def unsafeRun[E, A](zio: => ZIO[R, E, A]): A
+}
+```
+
+For example, if we wanted to integrate a ZIO workflow with a legacy unsafe code, we used to write something like this:
+
+```scala
```
Comment (adamgfraser) [inline docs/guides/migrate/migration-guide.md:746]: Do you want to do the migration table for `Runtime.unsafe***` here or in a separate PR?

**Review by adamgfraser — CHANGES_REQUESTED** [review]
Looks great! Just a few comments.

File: docs/guides/migrate/migration-guide.md (hunk 1 above)
Comment (khajavi) [reply docs/guides/migrate/migration-guide.md]: Weird change! Wondering how it comes to my commits :)
Fixed!

File: docs/guides/migrate/migration-guide.md (hunk 2 above)
Comment (khajavi) [reply docs/guides/migrate/migration-guide.md:738]: I agree, but some of them are based on ZIO 1.x, one of them is just a `diff` snippet based on ZIO 2.x, some of them are just simplified definitions of the API, and one another is based on scala 3. All of these are not type-safe. 

Fortunately, the final snippet has a type-safe mdoc modifier.

**Review by adamgfraser — APPROVED** [review]

File: docs/guides/migrate/migration-guide.md [hunk 4]
```diff
@@ -729,6 +729,105 @@ We deprecated the `Fiber.ID` and moved it to the `zio` package and called it the
...
+Unsafe.unsafeCompat { implicit unsafe =>
+  Runtime.default.unsafe.run(Console.printLine("Hello, World!"))
+}
+```
+
+In summary, here are the rules for migrating from ZIO 1.x to ZIO 2.x corresponding to the unsafe operators:
+
+|         | ZIO 1.0                    | ZIO 2.x                                                     |
+|---------|----------------------------|-------------------------------------------------------------|
+| Scala 2 | `zio.Runtime.unsafeRun(x)` | `Unsafe.unsafe{ implicit u => zio.Runtime.unsafe.run(x) }`  |
```
Comment (adamgfraser) [inline docs/guides/migrate/migration-guide.md:828]: @khajavi Would be good to add the other `unsafeRun` variants from ZIO 1.0 to this table.
